### PR TITLE
refactor(framework) Update `huggingface` template for `flwr new`

### PR DESCRIPTION
**Pull request by jafermarq**
Updates `HF` `flwr new` template.

needs: https://github.com/adap/flower/pull/4119

### CODE REVIEW

**Review by chongshenng — DISMISSED** [review]
Overall looks fine to me. The only minor comment is in `pyproject.toml`, whether we should set `prajjwal1/bert-tiny` as default and add a comment for users with access to larger GPUs, e.g.:

```
model-name = "prajjwal1/bert-tiny"  # If you have access to more GPU resources, 
                                    # you can try a larger model, such as "distilbert-base-uncased"
```

File: src/py/flwr/cli/new/templates/app/code/task.huggingface.py.tpl [hunk 1]
```diff
@@ -4,24 +4,23 @@ import warnings
 from collections import OrderedDict
 
 import torch
```
Comment (chongshenng) [inline src/py/flwr/cli/new/templates/app/code/task.huggingface.py.tpl:6]: It'd be good to only show the error logs from `transformers` and not warnings.
```suggestion
import torch
import transformers
```

File: src/py/flwr/cli/new/templates/app/code/task.huggingface.py.tpl [hunk 2]
```diff
@@ -4,24 +4,23 @@ import warnings
...
 
-from flwr_datasets import FederatedDataset
-from flwr_datasets.partitioner import IidPartitioner
-
-
 warnings.filterwarnings("ignore", category=UserWarning)
-DEVICE = torch.device("cpu")
-CHECKPOINT = "distilbert-base-uncased"  # transformer model checkpoint
+warnings.filterwarnings("ignore", category=FutureWarning)
+disable_progress_bar()
```
Comment (chongshenng) [inline src/py/flwr/cli/new/templates/app/code/task.huggingface.py.tpl:17]:
```suggestion
disable_progress_bar()
transformers.logging.set_verbosity_error()
```

**Review by chongshenng — DISMISSED** [review]
Tiny improvements to suppress `transformers` warnings.

File: src/py/flwr/cli/new/templates/app/pyproject.huggingface.toml.tpl [hunk 3]
```diff
@@ -29,10 +29,18 @@ clientapp = "$import_name.client_app:app"
...
 [tool.flwr.federations]
 default = "localhost"
 
 [tool.flwr.federations.localhost]
 options.num-supernodes = 10
+
+[tool.flwr.federations.localhost-gpu]
+options.num-supernodes = 10
+options.backend.client-resources.num-cpus = 2 # each ClientApp assumes to use 4CPUs
+options.backend.client-resources.num-gpus = 0.25 # at most 4 ClientApps will run in a given GPU
```
Comment (chongshenng) [inline src/py/flwr/cli/new/templates/app/pyproject.huggingface.toml.tpl:46]:
```suggestion
options.backend.client-resources.num-cpus = 4 # each ClientApp assumes to use 4CPUs
options.backend.client-resources.num-gpus = 0.25 # at most 4 ClientApps will run in a given GPU

```

**Review by danieljanes — APPROVED** [review]

**Review by danieljanes — APPROVED** [review]

**Pull request merged**